src/aux-records/XpController.spec.ts

```diff
@@ -0,0 +1,147 @@
+import { createTestControllers, createTestUser } from './TestUtils';
+import { MemoryAuthMessenger } from './MemoryAuthMessenger';
+import { AuthController } from './AuthController';
+import { MemoryStore } from './MemoryStore';
+import { GetXpUserResult, XpController } from '../aux-records/XpController';
+import { XpUser } from './XpStore';
+import { isOfXType } from './Utils';
+import { PromiseT } from './TypeUtils';
+
+/**
+ * XpController tests
+ */
+describe('XpController', () => {
+    //* BOCS: Init services
+    let services: ReturnType<typeof createTestControllers>;
+    let xpController: XpController;
+    let memoryStore: MemoryStore;
+    let authController: AuthController;
+    let authMessenger: MemoryAuthMessenger;
+    //* EOCS: Init services
+
+    /**
+     * (Re-)Initialize services required for tests
+     */
+    const initServices = () => {
+        services = createTestControllers();
+        xpController = new XpController({
+            xpStore: services.store,
+            authController: services.auth,
+            authStore: services.store,
+        });
+        memoryStore = services.store;
+        authController = services.auth;
+        authMessenger = services.authMessenger;
+    };
+
+    /**
+     * Runs before all tests (once)
+     */
+    beforeAll(async () => {
+        initServices();
+    });
```

Comment:
@LandonSiler Technically this isn't needed because the services are being initialized before each test.